### PR TITLE
Math Magicians - Events

### PR DESCRIPTION
**Pull request by ReshyResh**
# Changes 
  - Add event listener on buttons
  - Use state to get/make updates to components

# Screenshot

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
+          <Button type="but digit" value="9" onclick={this.calc} />
+          <Button type="but operator" value="x" onclick={this.calc} />
+          <Button type="but digit" value="4" onclick={this.calc} />
+          <Button type="but digit" value="5" onclick={this.calc} />
+          <Button type="but digit" value="6" onclick={this.calc} />
+          <Button type="but operator" value="-" onclick={this.calc} />
+          <Button type="but digit" value="1" onclick={this.calc} />
+          <Button type="but digit" value="2" onclick={this.calc} />
+          <Button type="but digit" value="3" onclick={this.calc} />
+          <Button type="but operator" value="+" onclick={this.calc} />
```
Comment (dasileker) [inline src/components/Calculator.js:46]: - [ ] Good job on the props, but I don't see the validation of your **digit or operator**, please add validation to your props

File: src/components/Results.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
-const Results = () => <p className="result">0</p>;
+import PropTypes from 'prop-types';
+
+const Results = ({ value }) => <p className="result">{value}</p>;
+
+Results.propTypes = {
+  value: PropTypes.number,
```
Comment (dasileker) [inline src/components/Results.js:6]: - [ ] Nice on validating the result value, but you missed to add isRequired.

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
```
Comment (dasileker) [inline src/logic/calculate.js:15]: - [ ] Please remove all commented codes in all your files.

File: src/logic/operate.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '÷') {
+    return one.div(two).toString();
+  }
+  if (operation === '%') {
```
Comment (dasileker) [inline src/logic/operate.js:18]: - [ ] The modulus operation is not working, I should see result of modulus 300 it should return 3 as result.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ReshyResh,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @ReshyResh  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**